test/IPCSocket.cpp

```diff
@@ -144,29 +144,21 @@ void RPCSession::test_rewindToBlock(size_t _blockNr)
 
 void RPCSession::test_mineBlocks(int _number)
 {
-	// Extremely complicated mechanism because sometimes the miner breaks and stops mining.
 	u256 startBlock = fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString()));
 	u256 currentBlock = startBlock;
 	u256 targetBlock = startBlock + _number;
-	cout << "A" << endl;
-	for (size_t tries = 0; tries < 3 && startBlock < targetBlock; ++tries)
+	rpcCall("test_mineBlocks", { _number.str() }, true);
+
+	//@TODO do not use polling - but that would probably need a change to the test client
+	for (size_t polls = 0; polls < 100; ++polls)
 	{
-		if (currentBlock == startBlock)
-		{
-			cout << "MINE" << endl;
-			rpcCall("test_mineBlocks", { (targetBlock - startBlock).str() }, true);
-		}
-		cout << "WOIT" << endl;
-		startBlock = currentBlock;
-		//@TODO do not use polling - but that would probably need a change to the test client
-		for (size_t polls = 0; polls < 10; ++polls)
-		{
-			currentBlock = fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString()));
-			if (currentBlock >= targetBlock)
-				return;
-			std::this_thread::sleep_for(chrono::milliseconds(1));
-		}
+		currentBlock = fromBigEndian<u256>(fromHex(rpcCall("eth_blockNumber").asString()));
```

Comment:
No need to assign that to an address - we should also delete the variable `currentBlock`